### PR TITLE
Image-Collection relationship change

### PR DESCRIPTION
**Pull request by koriaf**
Having single image in multiple collections is too messy so simplified it. Please note: after merged your local database is better to be removed and re-created again (recreated the migrations to save time).
The widget is updated as well to handle that

Hope it's still stable...

### CODE REVIEW

**Comment by monkeypants** [comment]
so after this merge, will the pulumi stack (Azure deployment) need to be destroyed and re-created to avoid slow migration? we can do that now but not for much longer, will soon have non-trivial data that will need to be preserved.

File: src/annotation/annotations_app/models/base.py [hunk 1]
```diff
@@ -50,34 +50,38 @@ class User(db.Model):
...
-images_to_collections_table = Table(
-    'images_to_collections_table',
-    db.metadata,
-    Column('image_id', ForeignKey('images.id'), primary_key=True),
-    Column('collection_id', ForeignKey('imagecollections.id'), primary_key=True)
-)
+def generate_short_id(len: int = 15):
+    """
+    Generates short random value, useful for unique IDs but shorter and more readable than UUID
+    """
```
Comment (monkeypants) [inline src/annotation/annotations_app/models/base.py:56]: OK, but UUID is fine too. Maybe a constraint limiting minimum length would be safer.

**Review by monkeypants — APPROVED** [review]

**Comment by RamParameswaran** [comment]
The DB migration changes caused breaking change on the deployed dev environment. 
I'll recreate the DB on Monday.

Going forward let's try not to change/delete existing migrations.

**Comment by monkeypants** [comment]
oh, maybe i shouldn't have merged this until after tomorrow's demo :/

I might need to do a localhost demo, we will soon have dev+prod envs and then yes, breaking migrations will definitely be a pain.

**Comment by RamParameswaran** [comment]
I will deploy a working version to the Azure environment today, that will be ready for demo tomorrow.

**Comment by koriaf** [comment]
and you can deploy any old commit, the thing about submodules is that we have more or less working widget version for each backend version

**Pull request merged**